saito-core/src/core/consensus/blockring.rs

```diff
@@ -200,6 +200,15 @@ impl BlockRing {
             }
         }
     }
+
+     pub fn get_block_hash_by_block_id(&self, block_id: u64) -> Option<SaitoHash> {
+        let insert_pos = (block_id % RING_BUFFER_LENGTH) as usize;
+        self.ring[insert_pos]
```

Comment:
this feels wrong. need some clarification on this

Reply:
as discussed on TG; 
**get_block_hash_by_block_id()** i added this method in blockring.rs to get block hash from block_id
Then using the returned block_hash i find block whose **safe_to_prune_transactions** need to be set true.